plugin.onedrive/addon.py

```diff
@@ -433,33 +435,38 @@ def report_error(e):
         xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
         list_item = xbmcgui.ListItem(addon.getLocalizedString(32053))
         params['action'] = 'open_simple_folder'
-        params['folder'] = 'view.recent'
-        url = base_url + '?' + urllib.urlencode(params)
-        xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
-        list_item = xbmcgui.ListItem(addon.getLocalizedString(32055))
-        params['action'] = 'open_drive_folder'
-        params['folder'] = 'special/photos'
-        url = base_url + '?' + urllib.urlencode(params)
-        xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
-        list_item = xbmcgui.ListItem(addon.getLocalizedString(32056))
-        params['folder'] = 'special/music'
-        url = base_url + '?' + urllib.urlencode(params)
-        xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
-        list_item = xbmcgui.ListItem(addon.getLocalizedString(32057))
-        params['action'] = 'open_simple_folder'
-        params['folder'] = 'shared'
+        params['folder'] = 'recent'
         url = base_url + '?' + urllib.urlencode(params)
         xbmcplugin.addDirectoryItem(addon_handle, url, list_item, True)
+        if content_type == 'image':
+            list_item = xbmcgui.ListItem(addon.getLocalizedString(32055))
+            params['action'] = 'open_drive_folder'
```

Comment:
can you elaborate on this? I already set `content_type` for the listitem which is given by kodi and I just resend it across all calls (line 443). `content_type` could be image, audio or video.
I saw `media_type` definition and looks like it must be set to movie, tvshow, season, episode, musicvideo, etc., but there is no way for the addon to know this specific type apart from the `content_type`.